src/hyp_map.rs

```diff
@@ -96,8 +96,44 @@ impl HypMapFixedRegion {
     }
 }
 
-/// Represents a virtual address region that must be allocated and populated to be mapped.
-struct HypMapPopulatedRegion {
+// Represents an area that must be reset to restore U-mode original state.
```

Comment:
So, why do we need to "reset" the task every time we want to run it? That's a pretty big runtime penalty to pay in order to use U-mode. If we consider the first candidate use case -- certificate parsing -- how many kBs of memory are we copying/zeroing on each attempt to use U-mode?

Instead, how about a program flow that looks something like this:

```
fn main() {
    init();
    let mut result = 0
    loop {
        let task = ecall_get_next_task(result);
        result = handle_task(task);
    }
}
```

Basically the U-mode task runs in a loop with an ECALL that returns the result of the task and suspends execution until we have another task to give it.

I don't disagree that we'd eventually want the ability to reset U-mode, e.g. in the event the U-mode task is totally borked, but doing it every time we use it seems excessive.

Reply:
The original discussion was on the following decision branch (_simplified_):

1. Single task with multiple functionalities or multiple tasks with single functionalities.

Single task was decided.

2. If single task, how do we ensure / at what level of paranoia we guarantee isolation inside it:
 a. We keep the umode stateful, and then just restart it if things go wrong.
 b. We keep the umode stateless, and we reset the state of umode as of its creation every time we run, providing context through an input area to be mapped/copied in when we run it -- keeping an eye on performance.

Consensus was somewhat on 'b', so this is what this code implement. 
'a' could be a viable option at this point (not requiring much change to the above code). And would possibly make it easier to just switch in the future to multiple task.

Reply:
I restructured the code in `umode.rs` to make clear a few points. In general, we don't have to reset every time we run, but every time we end the lifetime of an active task. Even in the previous code I could hold a active task for some time and execute `run()` multiple times.

This new iteration hopefully makes things clearer.

For example, we could have a `PerCpu::` field that holds a `Option<UmodeActiveTask>`. This will allow us to reset on `panic` and to keep a stateful `umode` without changing anything in this code.

Essentially this discussion must be had, but I hope to have made it orthogonal to this PR.

Reply:
Pushing implementation of reset in this iteration has clearly been a distracting issue. I have removed it from these changes. This PR is now strictly about running Umode until the first e-call.

Decision on when to reset the state is one that I am preparing (writing document as we speak, although possibly can be had in the github issue).